### PR TITLE
Feature: Python wheel-based SDK builds and publishing

### PR DESCRIPTION
**Pull request by t0yv0**
### Proposed changes

This feature enables wheel-based builds and PyPI publishing for the provider's Python SDK. It is part of a larger rollout across all providers https://github.com/pulumi/home/issues/2883

### Related issues (optional)

Fixes https://github.com/pulumi/pulumi-aws-apigateway/issues/88

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -95,14 +95,17 @@ gen_python_sdk::
 	cd provider/cmd/${CODEGEN} && go run . python ../../../sdk/python ${SCHEMA_PATH}
 	cp ${WORKING_DIR}/README.md sdk/python
 
+build_python_sdk:: PYPI_VERSION = $(shell pulumictl convert-version -l python -v "$(VERSION)")
 build_python_sdk:: gen_python_sdk
```
Comment (mjeffryes) [inline Makefile:99]: did you intend to drop the dependency on gen_python_sdk?

Comment (t0yv0) [reply Makefile:99]: Hm does this change drop a dependency? Let me try to rerun this from a clean work tree.

Comment (mjeffryes) [reply Makefile:99]: sorry no, I misread the diff. Makefiles are hard

Comment (t0yv0) [reply Makefile:99]: I think the dependency is still there.. When I run `make build_python_sdk` it generates it first. This is just a Make syntax we use elsewhere for exposing env variables to the target.

**Review by mjeffryes — APPROVED** [review]

**Pull request merged**